packages/components/src/settings-form/types.ts

```diff
@@ -0,0 +1,32 @@
+export interface StringToString {
+	[ key: string ]: string;
+}
+
+export interface Field {
+	id: string;
+	type: 'text' | 'password' | 'checkbox' | 'select';
+	title: string;
+	label: string;
+	description?: string;
+	default?: string;
+	class?: string;
+	css?: string;
+	options?: StringToString;
+	tip?: string;
+	value?: string;
+	placeholder?: string;
+}
+
+export interface GetInputPropsReturn {
```

Comment:
I think `Get` in this type name is confusing.

Reply:
I just named it that since it's describing the return value of the function `getInputProps`. Is there another convention that makes more sense here @samueljseay ?

Reply:
I think its better to describe it by what this return value actually is rather than typing it because you use it for the return value of a specific function.

Its hard for me to guess what this thing actually is. It looks like its maybe a JSX element or something

Reply:
> I think its better to describe it by what this return value actually is rather than typing it because you use it for the return value of a specific function.

Yup, I like that approach. I can see how you would benefit from using TS from the start of a project since you'd have defined data structures in mind, as opposed to just "whatever this function needs."